aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -80,11 +80,12 @@ static inline ScalarType combine_categories(ScalarType higher, ScalarType lower)
   // NOLINTNEXTLINE(bugprone-branch-clone)
   if(isComplexType(higher)) {
     return higher;
```

Comment:
which branch do we take for this case:
```py
a = torch.randn((2, 2), dtype=torch.float)
b = torch.tensor(1, dtype=torch.double)
print((a+b).dtype) #float
```

Reply:
For this case, we take

https://github.com/pytorch/pytorch/blob/e4cb9f8bf1ab0f41a3f75dba0ae012a693cc5182/aten/src/ATen/native/TypeProperties.cpp#L85-L86